File: hw/bonilla/U3/HW16MazeGenerator/HW16MazeGeneratorJs/model/Direction.js
export const Direction = {
  NORTH: { rowOffset: -1, colOffset: 0, name: 'NORTH' },
  SOUTH: { rowOffset: 1, colOffset: 0, name: 'SOUTH' },
  EAST:  { rowOffset: 0, colOffset: 1, name: 'EAST' },
  WEST:  { rowOffset: 0, colOffset: -1, name: 'WEST' },

  all() {
    return [this.NORTH, this.SOUTH, this.EAST, this.WEST];
  },

  opposite(dir) {
    switch (dir) {
      case this.NORTH: return this.SOUTH;
      case this.SOUTH: return this.NORTH;
      case this.EAST: return this.WEST;
      case this.WEST: return this.EAST;
    }
  }
};

File: hw/bonilla/U3/HW17MazeGenerator/HW16MazeGenerator/src/main/java/ec/edu/espe/hw16mazegenerator/model/Direction.java
package ec.edu.espe.hw16mazegenerator.model;

/**
 *
 * @author Bonilla David SoftCrafters
 */
public enum Direction {
    NORTH(-1, 0), SOUTH(1, 0), EAST(0, 1), WEST(0, -1);

    final int rowOffset;
    final int colOffset;

    Direction(int rowOffset, int colOffset) {
        this.rowOffset = rowOffset;
        this.colOffset = colOffset;
    }

    public static Direction opposite(Direction dir) {
        return switch (dir) {
            case NORTH ->
                SOUTH;
            case SOUTH ->
                NORTH;
            case EAST ->
                WEST;
            case WEST ->
                EAST;
        };
    }

    public int getRowOffset() {
        return rowOffset;
    }

    public int getColOffset() {
        return colOffset;
    }

}
